daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -647,22 +646,49 @@ private[lf] case class PartialTransaction(
   /** Close a try context, by catching an exception,
     * i.e. a exception was thrown inside the context, and the catch associated to the try context did handle it.
     */
-  @nowarn("msg=parameter value (exceptionType|exception) in method rollbackTry is never used")
   def rollbackTry(
-      exceptionType: Ast.Type,
-      exception: Value[Value.ContractId],
-  ): PartialTransaction =
+      ex: SValue.SException
+  ): PartialTransaction = {
+    import scala.Ordering.Implicits.infixOrderingOps
     context.info match {
       case info: TryContextInfo =>
         // TODO https://github.com/digital-asset/daml/issues/8020
         //  in the case of there being no children we can simple drop the entire rollback node.
         val rollbackNode = Node.NodeRollback(context.children.toImmArray)
+        if (oldestContainedVersion(rollbackNode) < TxVersion.minExceptions) {
+          throw DamlEUnhandledException(ex)
+        }
         copy(
           context = info.parent.addRollbackChild(info.nodeId, context.nextActionChildIdx),
           nodes = nodes.updated(info.nodeId, rollbackNode),
         ).resetActiveState(info.beginState)
       case _ => throw new RuntimeException("rollbackTry called in non-catch context")
     }
+  }
+
+  private def oldestContainedVersion(top: Node): TxVersion = {
```

Comment:
This is somewhat inefficient. You are retraversing everything under the rollback node. We could instead keep track of the minimum below each subtree as we build up the transaction and avoid retraversing anything.
Fine for this PR but I think we probably wan to revisit this later.